src/lib.rs

```diff
@@ -443,52 +439,43 @@ impl Shell {
     ///
     /// Environment of the [`Shell`] affects all commands spawned via this
     /// shell.
-    pub fn var<K: AsRef<OsStr>>(&self, key: K) -> Result<String> {
+    pub fn var(&self, key: impl AsRef<OsStr>) -> Result<String> {
         self._var(key.as_ref())
     }
     fn _var(&self, key: &OsStr) -> Result<String> {
         match self._var_os(key) {
-            Some(it) => it.into_string().map_err(VarError::NotUnicode),
+            Some(it) => match it.to_str() {
+                Some(it) => Ok(it.to_string()),
+                None => Err(VarError::NotUnicode(key.into())),
+            },
             None => Err(VarError::NotPresent),
         }
-        .map_err(|err| Error::new_var(err, key.to_os_string()))
+        .map_err(|err| Error::new_var(err, key.into()))
```

Comment:
We need to think through holistic set of names for env&path manipulation functions. Perhaps

```
current_dir
set_current_dir
with_current_dir

env_var
set_env_var
remove_env_var
with_env_var

env_vars

```

Notes:

* either we have `set_env_var` with option or we have set/remove pair
* in the latter case,  I don't think we need `without_env_var` --- this is a rare operation, let the caller explicitly clone and remove env var
* we need `os` versions as well
* ideally, we also need `clear_env`. But that's hard, because windows just doesn't work if you clear _entire_ env. So, we shouold probably have `clear_env` which on windows acutally leaves some of the vars intact, and `clear_env_fully` which makes it completely clear. No need to add this API right of the bat, but we should make sure it'll fit with our naming conventions otherwise. 
* env_vars probably wants to return an iterator, like the stdlib? Although I don't understand why std returns an iterator here, it holds a verctor internally anyway, and could have returned a vec. Need to make a decission betwen sticking with mirroring std, or trying to improve upon it.